cognite/extractorutils/uploader.py

```diff
@@ -237,8 +245,14 @@ def upload(self) -> None:
                         self.logger.error("Error in upload callback: %s", str(e))
 
             self.upload_queue.clear()
+            self.logger.info(f"Uploaded {self.upload_queue_size} rows in total")
             self.upload_queue_size = 0
 
+    @retry(exceptions=CogniteAPIError, tries=10, delay=5, max_delay=15, backoff=1.5)
+    def upload_batch(self, database: str, table: str, patch: List[Row]):
+        # Upload
+        self.cdf_client.raw.rows.insert(db_name=database, table_name=table, row=patch, ensure_parent=True)
```

Comment:
You could use the [`retry_call`](https://github.com/invl/retry#retry_call) function instead of separating out a single line to a dedicated function and adding the decorator

Reply:
I like function + decorator more for the purpose of explicit stack trace in case it fails.